java-jacoco-previous/pom.xml

```diff
@@ -0,0 +1,77 @@
+    <artifactId>java</artifactId>
+    <version>3.4-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>java-jacoco-previous</artifactId>
+  <description>
+    This module shades the required classes of previous version of JaCoCo to allow analysis of two JaCoCo binary format.
+  </description>
+  <dependencies>
+    <dependency>
+      <groupId>org.jacoco</groupId>
+      <artifactId>org.jacoco.core</artifactId>
+      <version>${jacoco.previous.version}</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <dependencyReducedPomLocation>target/dependency-reduced-pom.xml</dependencyReducedPomLocation>
+              <filters>
+                <filter>
+                  <artifact>*:*</artifact>
+                  <includes>
+                    <include>org/jacoco/core/data/ExecutionDataReader*</include>
+                    <include>org/jacoco/core/data/ExecutionDataWriter*</include>
+                    <include>org/jacoco/core/analysis/Analyzer*</include>
+                    <include>org/jacoco/core/analysis/CoverageNodeImpl*</include>
+                    <include>org/jacoco/core/internal/**</include>
+                  </includes>
+                </filter>
+              </filters>
+              <relocations>
+                <relocation>
+                  <pattern>org.jacoco.core.data.ExecutionDataReader</pattern>
+                  <shadedPattern>org.jacoco.previous.core.data.ExecutionDataReader</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.jacoco.core.data.ExecutionDataWriter</pattern>
+                  <shadedPattern>org.jacoco.previous.core.data.ExecutionDataWriter</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.jacoco.core.analysis.Analyzer</pattern>
+                  <shadedPattern>org.jacoco.previous.core.analysis.Analyzer</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.jacoco.core.analysis.CoverageNodeImpl</pattern>
+                  <shadedPattern>org.jacoco.previous.core.analysis.CoverageNodeImpl</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.jacoco.core.internal</pattern>
+                  <shadedPattern>org.jacoco.previous.core.internal</shadedPattern>
+                </relocation>
+              </relocations>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+</project>
```

Comment:
@benzonico I personally prefer newline at the end of all textual files, and we supposed to have an active rule for Java files, but unfortunately not for xml